torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -273,8 +273,9 @@ class TensorPipeAgent : public RpcAgent {
   // only if it isn't yet. It does exist for errors (setErrorIfNeeded) but, even
   // then, it ends up printing a log message, which may worry the user. To solve
   // both issues we use a separate atomic flag to know the status of the future.
-  struct AtomicFutureMessage {
-    FutureMessage futMsg;
+  struct AtomicJitFuture {
+    std::shared_ptr<JitFuture> jitFuture =
+        std::make_shared<JitFuture>(at::AnyClassType::get());
```

Comment:
This change (replacing a value with a shared_ptr to a value) is undoing a slight optimization I had added: instead of having two nested shared_ptrs (hence two allocations, and two separate control blocks, with two reference counts) the code was using a single allocation (for the AtomicFutureMessage) with a single control block, and was relying on the shared_ptr "aliasing" constructor to extract a shared_ptr to the FutureMessage field that however still used the former control block.

The merits of this optimization are arguable, and it also has downsides (it keeps the atomic_flag alive even once it's not needed anymore), thus I'm fine to remove it, I just wanted to make sure this was intentional.

Reply:
This wasn't intentional. Let me add that to my todo list, and see if I can bring it back in a followup PR.